src/components/RegisterForm.jsx

```diff
@@ -1,4 +1,160 @@
-import React from 'react';
+/* eslint-disable class-methods-use-this */
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Form, Button, Spinner } from 'react-bootstrap';
+import { registerUser, showErrors } from '../store/actions/registerActions';
+import validate from '../utils/registerFormValidator';
+import store from '../store/store';
+import { IS_LOADING, LOGIN } from '../store/actions/actionTypes';
 
-const RegisterForm = () => <div className="main-container">Register</div>;
+const { REACT_APP_FRONTEND_URL } = process.env;
+
+export class ConnectedRegisterForm extends Component {
+  constructor() {
+    super();
+    this.state = {
+      email: '',
+      username: '',
+      password1: '',
+      password2: '',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  swapForm() {
+    store.dispatch({ type: LOGIN });
+  }
+
+  handleFocus(event) {
+    event.target.classList.remove('error');
+  }
+
+  handleChange(event) {
+    this.setState({ [event.target.id]: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    let {
+      email, username,
+    } = this.state;
+    const { password1, password2 } = this.state;
+    email = email.trim();
+    username = username.trim();
+    const errors = validate(username, email, password1, password2);
+    if (Object.keys(errors).length === 0) {
+      store.dispatch({ type: IS_LOADING, payload: { isLoading: true } });
+      const payload = {
+        user: {
+          email,
+          username,
+          password: password2,
+          callback_url: REACT_APP_FRONTEND_URL,
+        },
+      };
+
+      const { RegisterUser } = this.props;
+      RegisterUser(payload);
+    } else {
+      store.dispatch(showErrors(errors));
+    }
+  }
+
+  render() {
+    const { errors, isLoading } = this.props;
+    return (
+      <div className="center">
+        <Form onSubmit={this.handleSubmit}>
+          <Form.Group>
+            <Form.Label>Email</Form.Label>
+            <Form.Text id="email-error" className="error-text">{errors.email}</Form.Text>
+            <Form.Control
+              className={errors.email ? 'error' : ''}
+              type="text"
+              id="email"
+              onChange={this.handleChange}
+              onFocus={this.handleFocus}
+            />
+          </Form.Group>
+          <Form.Group>
+            <Form.Label>Username</Form.Label>
+            <Form.Text id="username-error" className="error-text">{errors.username}</Form.Text>
+            <Form.Control
+              className={errors.username ? 'error' : ''}
```

Comment:
@ElMonstro, you do not need a conditional statement here. You can use `{errors.username && 'error'}` instead which may evaluate to `error` or false.